src/components/NotificationsAndModals/Dialog/Dialog.vue

```diff
@@ -0,0 +1,162 @@
+<template>
+  <transition name="fade">
+    <div
+      v-if="show"
+      class="pb-dialog layer-modal"
+    >
+      <div
+        class="dialog-container"
+        style="max-width: 500px"
+      >
+        <div style="margin-bottom: 8px;">
+          <div class="close-button-container">
+            <PbButton
+              class="close-button"
+              icon="fas fa-times"
+              @click.native="close"
+            />
+          </div>
+          <div class="header">
+            <slot name="icon-header" />
+            <h2
+              class="pb"
+              style="color: var(--color-gray-90);"
+            >
+              {{ title }}
+            </h2>
+          </div>
+          <h5
+            style="color: var(--color-gray-90);"
+            class="pb-light"
+          >
+            {{ subtitle }}
+          </h5>
+        </div>
+        <div style="padding: 16px 0 16px 0">
+          <slot name="main" />
+        </div>
+        <div class="dialog-actions">
+          <div>
+            <slot name="footer" />
+          </div>
+          <div>
+            <PbButton
+              color="gray-40"
+              button-style="regular"
+              @click.native="close"
+            >
+              Fechar
+            </PbButton>
+            <PbButton
+              :color="buttonColor"
+              button-style="background"
+              @click.native="close"
```

Comment:
create a custom event so the parent component knows whenever the user clicks the custom button or the "close".